src/content_sao.cpp

```diff
@@ -1349,6 +1358,38 @@ void PlayerSAO::setBreath(const u16 breath, bool send)
 		m_env->getGameDef()->SendPlayerBreath(this);
 }
 
+void PlayerSAO::hide()
+{
+	if (m_hidden)
+		return;
+
+	m_hidden = true;
+	if (!m_player)
+		return;
+
+	Server *server = m_env->getGameDef();
+	if (!server)
+		return;
+
+	server->sendFakeJoinLeaveMessage(m_player->getName(), false);
```

Comment:
sounds very hacky to me

Reply:
I agree (and mentioned this in my initial comment), do you or anyone else have a better and less hacky idea?

Reply:
Would it be less hacky to directly send a chat message?